src/main/java/org/apache/sling/i18n/impl/JcrResourceBundleProvider.java

```diff
@@ -791,4 +733,102 @@ public String toString() {
             return "Key(" + baseName + ", " + locale + ")";
         }
     }
+
+    /**
+     * Registry of the loaded <code>resource bundles</code> and the associated <code>service registrations</code>
+     * The <code>ResourceBundleRegistry</code> takes care of the registration/deregistration of the resource bundles as OSGi services.
+     * It stores the references to the registered resource bundles and to the associated service registrations.
+     */
+    private static class ResourceBundleRegistry {
+        private final Logger log = LoggerFactory.getLogger(getClass());
+        private final BundleContext bundleContext;
+        final AtomicBoolean isClosed = new AtomicBoolean(false);
+        private final Map<Key, Entry> bundleServiceRegistrations = new ConcurrentHashMap<>();
```

Comment:
I would rename the field to `registrations`. But it's fine either way 🙂

Reply:
agree, since it registry holds both bundles and services